spring-ws-core/src/main/java/org/springframework/ws/soap/saaj/SaajSoapMessageFactory.java

```diff
@@ -233,7 +235,19 @@ private MimeHeaders parseMimeHeaders(InputStream inputStream) throws IOException
 		MimeHeaders mimeHeaders = new MimeHeaders();
 		if (inputStream instanceof TransportInputStream) {
 			TransportInputStream transportInputStream = (TransportInputStream) inputStream;
-			for (Iterator<String> headerNames = transportInputStream.getHeaderNames(); headerNames.hasNext();) {
+			
+			//Remove duplicate keys
+			Iterator<String> headerNamesWithDuplicate = transportInputStream.getHeaderNames();
```

Comment:
AFAIK, SAAJ MimeHeaders is capable of dealing with multiple entries with the same header name. Not sure why we need to remove duplicates here.